demo/addons/isochroneAddon/isochroneAddon.js

```diff
@@ -0,0 +1,334 @@
+                if (input.value > 0) {
+                    distances.push(input.value*1000);
+                }
+            });
+
+            $("#metres_input").each(function(index,input) {
+                if (input.value > 0) {
+                    distances.push(input.value);
+                }
+            });
+
+            if (!_xy || (times.length === 0) && (distances.length === 0)) {
+                mviewer.alert("Isochrones : Il faut définir l'origine et au moins un temps de parcours ou une distance", "alert-info")
+                return;
+            }
+
+            /**
+             * adds all values from table together for one total time and total distance
+             * @param  {number} "totalTime", "totalDistance"
+             */
+
+            var totalTime = 0;
+            for(var t = 0; t<times.length; t++){
+              totalTime = totalTime + parseInt(times[t]);
+            }
+
+            var totalDistance = 0;
+            for(var t = 0; t<distances.length; t++){
+              totalDistance = totalDistance + parseInt(distances[t]);
+            }
+
+            /**
+             * gets other parameters for request
+             */
+
+            var mode = $(".selected.isochrone-mode").attr("data-mode");
+            var url = mviewer.customComponents["isochroneAddon"].config.options.isochroneUrl;
+
+            var dataParameters = {
+                            "location":_xy.join(","),
+                            "graphName":mode,
+                            "smoothing": true,
+                            "holes":true,
+                            "srs": "epsg:4326"   
+                        };
+
+            if( totalTime != 0 && totalDistance== 0){
+              dataParameters["time"]=totalTime;
+              dataParameters["method"]="time";
+            } else if( totalDistance != 0 && totalTime == 0){
+              dataParameters["distance"]=totalDistance;
+              dataParameters["method"]="distance";
+            }else{
+              mviewer.alert("Isochrones : Il faut définir soit un temps soit une distance, pas les deux", "alert-info")
+              return;
+            };
+            
+            /**
+             *Sends request, shows loading messages when request is sent, hides messages when result recieved
+             * @param  {} "#loading-isochrones"
+             * @param  {} .show
+             */
+
+            $("#loading-isochrones").show();
+                    $.ajax({
+                        type: "GET",
+                        url: url,
+                        crossDomain: true,
+                        data: dataParameters,
+                        dataType: "json",
+                        success: function (response) {
+                            console.log(response)
+                            _showResult(response);
+                        },
+                        error: function (request, status, error) {
+                            console.log(error);
```

Comment:
Maybe could use mviewer.alert() instead of console.log() to advert user ("call admin" or other msg).